tests/test_resource.py

```diff
@@ -233,28 +161,90 @@ def test_method(self):
+    def test_get_name_from__name_parameter(self) -> None:
         class XResource(Resource):
             _name = "test"
 
         assert XResource.get_name() == "test"
 
-    def test__get_name__uses_class_name_attribute_when_custom_name_is_not_defined(self):
+    def test_get_name_from__name__parameter(self) -> None:
         class XResource(Resource):
             pass
 
         assert XResource.get_name() == "xresource"
 
+    def test_get_all_possible_authz(self, sample_event) -> None:
+        azc = AuthzCollector()
+        azc.clean()
+
+        class XResource(Resource):
+            _name = "test_auth_collector"
+
+        # import pdb; pdb.set_trace()
+        res = XResource(sample_event)
+        assert res.get_authz_data() == {
+            "test_auth_collector": {
+                "possible_permissions": {},
+                "guest_permissions": {},
+            }
+        }
+
+
+class TestResourceAuthentication:
+    @pytest.fixture(autouse=True)
+    def setup_class(self, sample_event) -> None:
+        class XResource(Resource):
+            @add_route("/")
+            def test_method(self) -> None:
+                return Response("x")
+
+        self.res = XResource  # pylint: disable=attribute-defined-outside-init
+        self.event = sample_event  # pylint: disable=attribute-defined-outside-init
+
+    def test_unauthorized_when_authentication_not_configured(self) -> None:
+        resp = self.res({**self.event, "headers": {"authentication": "dummy"}})()
+        assert resp.status_code == HTTPStatus.UNAUTHORIZED
+
+    @patch.object(User, "__init__", return_value=None)
+    def test_user_loaded_when_cognito_authentication_configured_correctly(
+        self, load_user: MagicMock
+    ):
+        key = json.loads(env_mock["ALLOWED_PUBLIC_KEYS"])["keys"][0]
+        key_id = key["kid"]
+        authentication_token = jwt.encode({"username": "x"}, "", headers={"kid": key_id})
+
+        self.res({**self.event, "headers": {"authentication": authentication_token}})()
+        load_user.assert_called_once_with(authentication_token)
+
+    def test_unauthorized_when_jwt_header_lacks_kid(self) -> None:
+        authentication_token = jwt.encode({"foo": "bar"}, "")
+        resp = self.res({**self.event, "headers": {"authentication": authentication_token}})()
+        assert resp.status_code == HTTPStatus.UNAUTHORIZED
+
+    def test_unauthorized_when_no_matching_key_in_env_variable(self) -> None:
+        authentication_token = jwt.encode({"kid": "foobar"}, "")
+        resp = self.res({**self.event, "headers": {"authentication": authentication_token}})()
+        assert resp.status_code == HTTPStatus.UNAUTHORIZED
+
+    def test_unauthorized_when_jwt_header_malformed(self) -> None:
+        resp = self.res({**self.event, "headers": {"authentication": "12345"}})()
+        assert resp.status_code == HTTPStatus.UNAUTHORIZED
+
+    def test_unauthorized_when_authentication_not_configured_correctly(self) -> None:
```

Comment:
Here you are initializing and calling the instance of the Resource class - it is hard to guess what you mean byt `not_configured_correctly` 😉  It would be very good to show that more directly or write a short comment about it.